### PR TITLE
Applying Unified Model and miscellaneous TypeSpec changes to Java SDK

### PR DESCRIPTION
**Pull request by NickKouds**
# Description
Applied Unified Model TypeSpec changes to Batch SDK

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/batch/batch-typespec/routes.tsp [hunk 1]
```diff
@@ -67,6 +93,8 @@ API.
 }
 
 #suppress "@azure-tools/typespec-azure-core/casing-style" "The names of Model types must use camelCase"
+#suppress "@azure-tools/typespec-azure-core/no-rpc-path-params" "Operations defined using RpcOperation should not have path parameters."
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Operation 'Exists' should be defined using a signature from the Azure.Core namespace."
```
Comment (dpwatrous) [inline sdk/batch/batch-typespec/routes.tsp:97]: Can we suppress this kind of error globally?

File: sdk/batch/batch-typespec/routes.tsp [hunk 2]
```diff
@@ -67,6 +93,8 @@ API.
 }
 
 #suppress "@azure-tools/typespec-azure-core/casing-style" "The names of Model types must use camelCase"
+#suppress "@azure-tools/typespec-azure-core/no-rpc-path-params" "Operations defined using RpcOperation should not have path parameters."
```
Comment (dpwatrous) [inline sdk/batch/batch-typespec/routes.tsp:96]: Is the path parameter needed here for technical reasons or is it a style preference?

File: sdk/batch/batch-typespec/routes.tsp (hunk 1 above)
Comment (NickKouds) [reply sdk/batch/batch-typespec/routes.tsp:97]: Unfortunately not - Don't think in our experience TypeSpec typically offers a way to globally surpress something. Either hopefully once the TypeSpec team has a solution to the ResourceExists template, we won't need to surpress this anymore.

**Review by dpwatrous — APPROVED** [review]

**Pull request merged**